File: ruby/tests/test-atomicweight.rb
require "xraylib"
require "test/unit"

class TestAtomicWeight < Test::Unit::TestCase
	def test_Fe
		weight = Xraylib.AtomicWeight(26)
		assert_in_delta(weight, 55.850, 1E-6)
	end

	def test_U
		weight = Xraylib.AtomicWeight(92)
		assert_in_delta(weight, 238.070, 1E-6)
	end

	def test_bad_Z
		assert_raise(ArgumentError) {
			Xraylib.AtomicWeight(185)
		}
		assert_raise(ArgumentError) {
			Xraylib.AtomicWeight()
		}
		assert_raise(TypeError) {
			Xraylib.AtomicWeight(nil)
		}
		assert_raise(TypeError) {
			Xraylib.AtomicWeight("hwoefhhfowfhwfh")
		}
		assert_raise(TypeError) {
			Xraylib.AtomicWeight("26")
		}
		assert_raise(ArgumentError) {
			Xraylib.AtomicWeight(26, -5)
		}
	end
end

File: python/tests/test-atomicweight.py
import os
if "EXTRA_DLL_SEARCH_PATHS" in os.environ and hasattr(os, "add_dll_directory"):
    for path in os.environ["EXTRA_DLL_SEARCH_PATHS"].split(os.pathsep):
        os.add_dll_directory(path)
import unittest
import xraylib


class TestAtomicWeight(unittest.TestCase):
    def test_Fe(self):
        weight = xraylib.AtomicWeight(26)
        self.assertAlmostEqual(weight, 55.850)

    def test_U(self):
        weight = xraylib.AtomicWeight(92)
        self.assertAlmostEqual(weight, 238.070)

    def test_bad_Z(self):
        with self.assertRaises(ValueError):
            width = xraylib.AtomicWeight(185)

if __name__ == '__main__':
    unittest.main(verbosity=2)

